### PR TITLE
Updates for Stormbood changes

### PR DESCRIPTION
**Pull request by Ermad**
I have been implementing some of the changes for Stormblood. So far just the new recipes and actions, but its a start that makes it useable even if not entirely accurate.

This it self isn't ready to go yet (still need to link up all the new xivdb tooltips for example), but thought would show the changes I have so far.

### CODE REVIEW

**Comment by doxxx** [comment]
Thanks for looking at this. I assume you added the new recipes using https://github.com/doxxx/lodestone-recipe-db-scraper?

**Comment by doxxx** [comment]
I can take a look at fixing the XIVDB tooltips separately. Looks like the API has changed again -_-

**Comment by doxxx** [comment]
No, the API hasn't changed, it's just not returning data for some of the actions -- probably the new Stormblood actions. Unfortunately, the tooltip service doesn't handle that correctly, so I'll fix that first and then try to figure out why it's not returning data for those actions.

**Comment by Ermad** [comment]
Yea, I made the small changes needed and used lodestone-recipe-db-scraper. 

For the tooltips, I meant adding all the right spell ids in services/actions.js. I haven't noticed an issue with the tooltips otherwise.

**Comment by doxxx** [comment]
I've fixed the tooltip service on master so that missing data in the response from XIVDB won't break it completely -- it will still get all the tooltips that XIVDB has data for.

File: app/js/ffxivcraftmodel.js [hunk 1]
```diff
@@ -1731,7 +1795,17 @@ var LevelTable = {
     57: 142, // 142
     58: 145, // 145
     59: 148, // 148
-    60: 150  // 150
+    60: 150, // 150
+    61: 260,
```
Comment (doxxx) [inline app/js/ffxivcraftmodel.js]: I think you'll need to add a block for crafterLevel >= 260 in the calculateBaseProgressIncrease function. You can start with the contents from the crafterLevel >= 120 block and adjust as necessary.

**Comment by Ermad** [comment]
I haven't had the time yet to collect progress/quality data to update the formula with though. So I haven't started tackling that part yet.

**Comment by doxxx** [comment]
I've updated the lodestone-recipe-db-scraper to be able to merge in the Chinese recipe names provided by @kurax. Please see the README for updated instructions on how to run the tool.

**Comment by doxxx** [comment]
I'll be merging in #138 today. I'm hoping it won't cause conflicts for your branch, but if it does -- sorry!

**Comment by doxxx** [comment]
Yeah, sorry, merging #138 has caused some conflicts but they look simple to resolve -- taking your new generated recipedb files over the merged changes.

**Comment by doxxx** [comment]
Can you infer whether a particular crafter class is a specialist by the presence of enabled Specialist actions?

**Comment by Ermad** [comment]
It's possible, but I don't think it would be very accurate. It relies on users to set those actions accurately, which they may not.

With the new specialist effects (one being passive and active for every craft), I think being more accurate and explicit about it is worth the extra control.

File: app/js/solver/worker.js [hunk 2]
```diff
@@ -248,7 +249,8 @@ function postProgress(gen, maxGen, best, synthNoConditions) {
         hqPercent: hqPercentFromQuality(currentState.qualityState / synthNoConditions.recipe.maxQuality * 100),
         feasible: violations.progressOk && violations.durabilityOk && violations.cpOk && violations.trickOk && violations.reliabilityOk,
         violations: violations,
-        condition: currentState.condition
+        condition: currentState.condition,
+        bonusMaxCp: finalState.bonusMaxCp
```
Comment (doxxx) [inline app/js/solver/worker.js]: `finalState` should be `currentState`.

**Comment by Ermad** [comment]
I think this should be ready to go. The updated quality/progress formulas have been pretty accurate so far.

**Comment by doxxx** [comment]
Thank you very much for this, Ermad! I've published this and some other changes to the beta site.

**Comment by Ermad** [comment]
Thanks for the making the tool, I have used it so much in my crafting career.

**Pull request merged**